src/test/java/ch/naviqore/utils/cache/EvictionCacheTest.java

```diff
@@ -0,0 +1,139 @@
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class EvictionCacheTest {
+
+    public static final int CACHE_SIZE = 3;
+    public static final int CONCURRENT_ACCESS_COUNT = 1000;
+
+    @Nested
+    class LRU {
+
+        private EvictionCache<String, String> lruCache;
+
+        @BeforeEach
+        void setUp() {
+            lruCache = new EvictionCache<>(CACHE_SIZE, EvictionCache.Strategy.LRU);
+        }
+
+        @Test
+        void shouldComputeIfAbsent() {
+            assertThat(lruCache.isCached("a")).isFalse();
+            lruCache.computeIfAbsent("a", () -> "apple");
+            assertThat(lruCache.isCached("a")).isTrue();
+            assertThat(lruCache.computeIfAbsent("a", () -> "apple")).isEqualTo("apple");
+        }
+
+        @Test
+        void shouldRemoveLeastRecentlyUsed() {
+            lruCache.computeIfAbsent("a", () -> "apple");
+            lruCache.computeIfAbsent("b", () -> "banana");
+            lruCache.computeIfAbsent("c", () -> "cherry");
+
+            // access "a" again to make it recently used
+            lruCache.computeIfAbsent("a", () -> "apple");
+            // this should evict "b"
+            lruCache.computeIfAbsent("d", () -> "date");
+
+            assertThat(lruCache.isCached("b")).isFalse();
+            assertThat(lruCache.isCached("a")).isTrue();
+            assertThat(lruCache.isCached("c")).isTrue();
+            assertThat(lruCache.isCached("d")).isTrue();
+        }
+
+        @Test
+        void shouldClear() {
+            lruCache.computeIfAbsent("a", () -> "apple");
+            lruCache.computeIfAbsent("b", () -> "banana");
+            lruCache.computeIfAbsent("c", () -> "cherry");
+
+            lruCache.clear();
+
+            assertThat(lruCache.isCached("a")).isFalse();
+            assertThat(lruCache.isCached("b")).isFalse();
+            assertThat(lruCache.isCached("c")).isFalse();
+        }
+
+        @Test
+        void shouldHandleConcurrentAccess() throws InterruptedException, ExecutionException, TimeoutException {
+            runConcurrentAccess(lruCache);
+
+            assertThat(lruCache.getNumberOfEntries()).isEqualTo(CACHE_SIZE);
+        }
+    }
+
+    @Nested
+    class MRU {
+
+        private EvictionCache<String, String> mruCache;
+
+        @BeforeEach
+        void setUp() {
+            mruCache = new EvictionCache<>(CACHE_SIZE, EvictionCache.Strategy.MRU);
+        }
+
+        @Test
+        void shouldComputeIfAbsent() {
+            assertThat(mruCache.isCached("a")).isFalse();
+            mruCache.computeIfAbsent("a", () -> "apple");
+            assertThat(mruCache.isCached("a")).isTrue();
+            assertThat(mruCache.computeIfAbsent("a", () -> "apple")).isEqualTo("apple");
```

Comment:
again, () -> "avocado" should also work here, right?

Reply:
Yes, see [6633abc](https://github.com/naviqore/public-transit-service/pull/52/commits/6633abcbbb43b4737067cf923fd0773e04228153)